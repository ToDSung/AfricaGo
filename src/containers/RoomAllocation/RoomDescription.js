import React from 'react';

import PropTypes from 'prop-types';

const RoomDescription = ({ title = 'title', subtitle = '' }) => (
  <div className='my-2 flex flex-col'>
    <span>{title} </span>
    <span className='text-gray-400'>{subtitle}</span>
  </div>
);

RoomDescription.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
};

export default RoomDescription;
