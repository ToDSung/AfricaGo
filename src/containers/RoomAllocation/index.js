import React, { useState } from 'react';

import CustomInputNumber from '../../components/CustomInputNumber';

const RoomAllocation = () => {
  const [value, setValue] = useState(0);

  const handleChange = event => {
    setValue(event.target.value);
  };

  const handleBlur = event => {
    console.log(event);
  };

  return (
    <CustomInputNumber
      min={0}
      max={4}
      value={value}
      onChange={handleChange}
      onBlur={handleBlur}
    />
  );
};

export default RoomAllocation;
