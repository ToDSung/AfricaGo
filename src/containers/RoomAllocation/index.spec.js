/**
 * @jest-environment jsdom
 */

import React from 'react';
import { act, create } from 'react-test-renderer';

import { fireEvent, render } from '@testing-library/react';

import RoomAllocation from '.';

test('should RoomAllocation render', () => {
  const component = create(
    <RoomAllocation
      guest={10}
      room={3}
      onChange={result => console.log(result)}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => {
    component.update(
      <RoomAllocation
        guest={3}
        room={3}
        onChange={result => console.log(result)}
      />
    );
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should RoomAllocation calculate correctly', () => {
  const mockHandleChange = jest.fn();
  const { getAllByRole } = render(
    <RoomAllocation guest={10} room={3} onChange={mockHandleChange} />
  );

  const PlusNodes = getAllByRole(/Plus/i);
  const minusNodes = getAllByRole(/Minus/i);

  fireEvent.click(minusNodes[0]);
  expect(mockHandleChange).toHaveBeenCalledTimes(0);

  fireEvent.click(PlusNodes[0]);
  expect(mockHandleChange).toHaveBeenCalledTimes(1);
  expect(mockHandleChange).toHaveBeenCalledWith([
    {
      adult: 2,
      child: 0,
    },
    {
      adult: 1,
      child: 0,
    },
    {
      adult: 1,
      child: 0,
    },
  ]);

  fireEvent.click(minusNodes[0]);
  expect(mockHandleChange).toHaveBeenCalledTimes(2);
  expect(mockHandleChange).toHaveBeenCalledWith([
    {
      adult: 1,
      child: 0,
    },
    {
      adult: 1,
      child: 0,
    },
    {
      adult: 1,
      child: 0,
    },
  ]);

  [...Array(10)].forEach(() => {
    fireEvent.click(PlusNodes[0]);
  });
  expect(mockHandleChange).toHaveBeenCalledTimes(5);
  expect(mockHandleChange).toHaveBeenCalledWith([
    {
      adult: 4,
      child: 0,
    },
    {
      adult: 1,
      child: 0,
    },
    {
      adult: 1,
      child: 0,
    },
  ]);

  [...Array(10)].forEach(() => {
    fireEvent.click(PlusNodes[3]);
  });
  expect(mockHandleChange).toHaveBeenCalledTimes(8);
  expect(mockHandleChange).toHaveBeenCalledWith([
    {
      adult: 4,
      child: 0,
    },
    {
      adult: 1,
      child: 3,
    },
    {
      adult: 1,
      child: 0,
    },
  ]);

  [...Array(10)].forEach(() => {
    fireEvent.click(PlusNodes[5]);
  });
  expect(mockHandleChange).toHaveBeenCalledTimes(9);
  expect(mockHandleChange).toHaveBeenCalledWith([
    {
      adult: 4,
      child: 0,
    },
    {
      adult: 1,
      child: 3,
    },
    {
      adult: 1,
      child: 1,
    },
  ]);
});
