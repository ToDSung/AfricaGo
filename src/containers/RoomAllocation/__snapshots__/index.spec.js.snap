// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should RoomAllocation render 1`] = `
<div
  className="w-96 p-2"
>
  <h5>
    住家人數： 
    10
    人 / 
    3
    房
  </h5>
  <div
    className="flex items-center justify-center rounded border text-base my-2 mr-4 h-10 text-gray-600 border-amber-500 "
  >
    尚未分配人數: 
    7
    人
  </div>
  <div>
    房間：
    1
    人
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          大人
           
        </span>
        <span
          className="text-gray-400"
        >
          年齡 20+
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-sky-600 focus:border-2
            "
          onBlur={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          1
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl   cursor-pointer border-sky-600  text-sky-600"
          onClick={[Function]}
          onMouseDown={[Function]}
          role="plus"
        >
          +
        </div>
      </div>
    </div>
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          小孩
           
        </span>
        <span
          className="text-gray-400"
        >
          
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-sky-600 focus:border-2
            "
          onBlur={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          0
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl   cursor-pointer border-sky-600  text-sky-600"
          onClick={[Function]}
          onMouseDown={[Function]}
          role="plus"
        >
          +
        </div>
      </div>
    </div>
  </div>
  <div>
    房間：
    1
    人
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          大人
           
        </span>
        <span
          className="text-gray-400"
        >
          年齡 20+
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-sky-600 focus:border-2
            "
          onBlur={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          1
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl   cursor-pointer border-sky-600  text-sky-600"
          onClick={[Function]}
          onMouseDown={[Function]}
          role="plus"
        >
          +
        </div>
      </div>
    </div>
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          小孩
           
        </span>
        <span
          className="text-gray-400"
        >
          
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-sky-600 focus:border-2
            "
          onBlur={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          0
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl   cursor-pointer border-sky-600  text-sky-600"
          onClick={[Function]}
          onMouseDown={[Function]}
          role="plus"
        >
          +
        </div>
      </div>
    </div>
  </div>
  <div>
    房間：
    1
    人
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          大人
           
        </span>
        <span
          className="text-gray-400"
        >
          年齡 20+
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-sky-600 focus:border-2
            "
          onBlur={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          1
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl   cursor-pointer border-sky-600  text-sky-600"
          onClick={[Function]}
          onMouseDown={[Function]}
          role="plus"
        >
          +
        </div>
      </div>
    </div>
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          小孩
           
        </span>
        <span
          className="text-gray-400"
        >
          
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-sky-600 focus:border-2
            "
          onBlur={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          0
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl   cursor-pointer border-sky-600  text-sky-600"
          onClick={[Function]}
          onMouseDown={[Function]}
          role="plus"
        >
          +
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should RoomAllocation render 2`] = `
<div
  className="w-96 p-2"
>
  <h5>
    住家人數： 
    3
    人 / 
    3
    房
  </h5>
  <div
    className="flex items-center justify-center rounded border text-base my-2 mr-4 h-10 text-gray-600 border-gray-600 "
  >
    尚未分配人數: 
    7
    人
  </div>
  <div>
    房間：
    1
    人
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          大人
           
        </span>
        <span
          className="text-gray-400"
        >
          年齡 20+
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-gray-600 text-gray-600
            "
        >
          1
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl  border-gray-600 text-gray-600"
          role="plus"
        >
          +
        </div>
      </div>
    </div>
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          小孩
           
        </span>
        <span
          className="text-gray-400"
        >
          
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-gray-600 text-gray-600
            "
        >
          0
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl  border-gray-600 text-gray-600"
          role="plus"
        >
          +
        </div>
      </div>
    </div>
  </div>
  <div>
    房間：
    1
    人
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          大人
           
        </span>
        <span
          className="text-gray-400"
        >
          年齡 20+
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-gray-600 text-gray-600
            "
        >
          1
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl  border-gray-600 text-gray-600"
          role="plus"
        >
          +
        </div>
      </div>
    </div>
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          小孩
           
        </span>
        <span
          className="text-gray-400"
        >
          
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-gray-600 text-gray-600
            "
        >
          0
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl  border-gray-600 text-gray-600"
          role="plus"
        >
          +
        </div>
      </div>
    </div>
  </div>
  <div>
    房間：
    1
    人
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          大人
           
        </span>
        <span
          className="text-gray-400"
        >
          年齡 20+
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-gray-600 text-gray-600
            "
        >
          1
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl  border-gray-600 text-gray-600"
          role="plus"
        >
          +
        </div>
      </div>
    </div>
    <div
      className="grid grid-cols-2"
    >
      <div
        className="my-2 flex flex-col"
      >
        <span>
          小孩
           
        </span>
        <span
          className="text-gray-400"
        >
          
        </span>
      </div>
      <div
        className="flex gap-2 p-2"
      >
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none text-4xl border-gray-600 text-gray-600"
          role="minus"
        >
          -
        </div>
        <div
          className="flex items-center justify-center rounded border text-base 
              relative h-12 w-12 outline-none 
              border-gray-600 text-gray-600
            "
        >
          0
        </div>
        <div
          className="flex items-center justify-center rounded border text-base h-12 w-12  select-none  text-4xl  border-gray-600 text-gray-600"
          role="plus"
        >
          +
        </div>
      </div>
    </div>
  </div>
</div>
`;
